src/shared/vscode/window.ts

```diff
@@ -136,3 +142,15 @@ class DefaultWindow implements Window {
         return vscode.window.showSaveDialog(options)
     }
 }
+
+/**
+ * Outputs progress message to an output channel as well as reporting progress through normal means.
+ * @param progress vscode.Progress object generated through DefaultWindow
+ * @param entry ProgressEntry to output to output channel and progress
+ */
+export function report(progress: vscode.Progress<ProgressEntry>, entry: ProgressEntry) {
+    if (entry.message) {
+        ext.outputChannel.appendLine(entry.message)
+    }
+    progress.report(entry)
```

Comment:
`withProgress()` already has access to `progress`. Why force all the callers to import this module function? Would it make sense to do it inside `withProgress()` instead?

Reply:
I'll see what I can do. I'd rather not muck up the return object but it would make it neater on the other end.

Reply:
Done. The newer version works as far as I can tell.